mitodl/micromasters
# Exams: process exam results

Based off of #2994 
----
Best way to see the true diff of this PR is [here](https://github.com/mitodl/micromasters/compare/nl/2896-result-file-audit-trail...nl/2791-process-exam-results)


#### What are the relevant tickets?
Fixes #2791 

#### What's this PR do?
Adds support to process exam results

#### How should this be manually tested?
The easiest way to test this is to acquire a production result file (ping me for this) and modify your `ExamAuthorization` record for your user to match the result record in that file. Then run:
```python
from exams.pearson.download import ArchivedResponseProcessor
ArchivedResponseProcessor().process_zip('result.zip')
```
The method should return `(True, [])`.

Verify that this creates a `grades.models.ProctoredExamGrade` record with data matching what was in the exam file within the zip. Also verify that your `ExamAuthorization.exam_taken == True`.

#### Where should the reviewer start?
exams/pearson/download.py (`process_exam_file()`)

#### What GIF best describes this PR or how it makes you feel?
![Exams](http://media1.giphy.com/media/R5fsUG6AIJGBa/giphy.gif)
